lib/live_debugger_refactor/api/traces_storage.ex

```diff
@@ -0,0 +1,391 @@
+defmodule LiveDebuggerRefactor.API.TracesStorage do
+  @moduledoc """
+  API for managing traces storage. In order to properly use invoke `init/0` at the start of application.
+  It uses Erlang's ETS (Erlang Term Storage).
+  """
+
+  alias LiveDebugger.Structs.Trace
+  alias LiveDebugger.CommonTypes
+
+  @type ets_continuation() :: term()
+  @typedoc """
+  Pid is used to store mapping to table references.
+  It identifies ETS tables managed by EtsTableServer.
+  """
+  @type ets_table_id() :: pid()
+  @type table_identifier() :: ets_table_id() | reference()
+
+  @callback init() :: :ok
+  @callback insert(Trace.t()) :: true
+  @callback insert(table_ref :: reference(), Trace.t()) :: true
+  @callback get_by_id(table_identifier(), trace_id :: integer()) :: Trace.t() | nil
+  @callback get(table_identifier(), opts :: keyword()) ::
+              {[Trace.t()], ets_continuation()} | :end_of_table
+  @callback clear(table_identifier(), node_id :: pid() | CommonTypes.cid() | nil) :: true
+  @callback clear(table_identifier()) :: true
+  @callback get_table(ets_table_id()) :: reference()
+  @callback delete_table(table_identifier()) :: boolean()
+  @callback get_all_tables() :: [reference()]
```

Comment:
I think some of these functions can raise errors (e.g. when you provide incorrect table ref. For such functions it is recommended to use `bang!` convention https://hexdocs.pm/elixir/main/naming-conventions.html#trailing-bang-foo